### ISSUE TITLE
`‎FTSCheckpoint.load_state_dict` panics when `epoch_transitions_only=True` and `FTSEarlyStopping` is not attached

### ISSUE THREAD

**Issue by DimitrisMantas** [speediedan/finetuning-scheduler#29]
### To Reproduce

`‎FTSCheckpoint.load_state_dict` requires that `FTSEarlyStopping` is attached to the `Trainer`, but this callback is not required when `epoch_transitions_only=True`, resulting in checkpoints to not be loadable under otherwise valid configurations.

### Expected behavior

Something like this:

```python
if (
    self.finetuningscheduler_callback.curr_depth > self.best_ckpt_depth
    and not self.finetuningscheduler_callback.epoch_transitions_only
    and isinstance(early_stopping, finetuning_scheduler.FTSEarlyStopping)
):
    early_stopping.wait_count = 0
```

### Environment

* CUDA:
        - GPU:
                - NVIDIA A10G
        - available:         True
        - version:           12.8
* Packages:
        - finetuning-scheduler: 2.10.0
        - numpy:             2.4.2
        - pyTorch_debug:     False
        - pyTorch_git_version: 449b1768410104d3ed79d3bcfe4ba1d65c7f22c0
        - pyTorch_version:   2.10.0+cu128
        - pytorch-lightning: 2.6.0
        - tqdm:              4.67.3
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.11.15
        - version:           #7~24.04.1-Ubuntu SMP Thu Jan 22 21:04:49 UTC 2026

**Comment by speediedan** [comment]
Thanks for reporting this @DimitrisMantas, and apologies for the delayed response — I've had limited bandwidth for Fine-Tuning Scheduler these past few months.

You identified the constraint exactly right: an `FTSEarlyStopping` callback shouldn't be required when the schedule is configured with `epoch_transitions_only=True`, since FTS deliberately doesn't attach one in that mode.

The actual blocking line turned out to be slightly different from the one your snippet targets, so I wanted to flag it in case it helps anyone hitting this. The `wait_count` reset was already guarded on `epoch_transitions_only` — the failure came from an unconditional `assert isinstance(..., FTSEarlyStopping)` at the very top of `load_state_dict`, which ran before any of that logic and so failed on every load in this configuration. I've moved the assertion inside the branch that actually needs the callback, which keeps it meaningful as an invariant on that path while letting valid `epoch_transitions_only` checkpoints load.

I've added regression tests for both the `epoch_transitions_only` reload path and — to make sure the fix didn't over-correct — the case where an `FTSEarlyStopping` *is* attached and the `wait_count` reset should still happen.

Thanks again for catching this and for the time you put into both reports. Feel free to reach out anytime if you run into anything else or want to share more about your use case. Best of luck with your work!

Merged to `main` in #31. This will ship in **2.13.0** — note the version jump: PyTorch 2.11, 2.12 and 2.13 all shipped while FTS was between releases, so rather than back-fill releases nobody could have depended on, FTS is resuming its PyTorch-aligned cadence with a single catch-up release. Since 2.13.0 keeps a minimum supported PyTorch of 2.7.0, every version an intervening release would have targeted remains supported.
